engine/apps/mattermost/tasks.py

```diff
@@ -89,3 +93,76 @@ def on_alert_group_action_triggered_async(log_record_id):
+
+    try:
+        user = User.objects.get(pk=user_pk)
+        alert_group = AlertGroup.objects.get(pk=alert_group_pk)
+        notification_policy = UserNotificationPolicy.objects.get(pk=notification_policy_pk)
+        mattermost_messsage = alert_group.mattermost_messages.get(message_type=MattermostMessage.ALERT_GROUP_MESSAGE)
+    except User.DoesNotExist:
+        logger.warning(f"User {user_pk} is not found")
+        return
+    except AlertGroup.DoesNotExist:
+        logger.warning(f"Alert group {alert_group_pk} is not found")
+        return
+    except UserNotificationPolicy.DoesNotExist:
+        logger.warning(f"UserNotificationPolicy {notification_policy_pk} is not found")
+        return
+    except MattermostMessage.DoesNotExist as e:
+        if notify_user_about_alert_async.request.retries >= 10:
+            logger.error(
+                f"Alert group mattermost message is not created {alert_group_pk}. Hence stopped retrying for user notification"
+            )
+            return
+        else:
+            raise e
+
+    mattermost_channel = MattermostChannel.get_channel_for_alert_group(alert_group=alert_group)
+    if not mattermost_channel:
+        logger.error(f"Mattermost channel not found for user notification {user_pk}")
+        return
+
+    templated_alert = AlertGroupMattermostRenderer(alert_group).alert_renderer.templated_alert
+
+    if not hasattr(user, "mattermost_user_identity"):
+        message = "{}\nTried to invite {} to look at the alert group. Unfortunately {} is not in mattermost.".format(
+            templated_alert.title, user.username, user.username
+        )
+
+        UserNotificationPolicyLogRecord(
+            author=user,
+            type=UserNotificationPolicyLogRecord.TYPE_PERSONAL_NOTIFICATION_FAILED,
+            notification_policy=notification_policy,
+            alert_group=alert_group,
+            reason="User is not in Mattermost",
+            notification_step=notification_policy.step,
+            notification_channel=notification_policy.notify_by,
+            notification_error_code=UserNotificationPolicyLogRecord.ERROR_NOTIFICATION_IN_MATTERMOST_USER_NOT_IN_MATTERMOST,
+        ).save()
+    else:
+        message = "{}\nInviting {} to look at the alert group.".format(
+            templated_alert.title, user.mattermost_user_identity.mention_username
+        )
+
+    payload = {"root_id": mattermost_messsage.post_id, "message": message}
+
+    try:
+        client = MattermostClient()
+        mattermost_post = client.create_post(channel_id=mattermost_channel.channel_id, data=payload)
+    except MattermostAPITokenInvalid:
+        logger.error(f"Mattermost API token is invalid could not create post for alert {alert_group_pk}")
+    except MattermostAPIException as ex:
+        logger.error(f"Mattermost API error {ex}")
+        if ex.status not in [status.HTTP_401_UNAUTHORIZED]:
+            raise ex
+    else:
+        MattermostMessage.create_message(
```

Comment:
I don't think we need to track this message? (we won't be updating it in any way)
What we will need though is a success `UserNotificationPolicyLogRecord` entry to have a record the notification was sent (and a failed one in case of the exceptions above; we need to make sure there is a `failed` or `success` log record for each `triggered` one)